src/main/java/frc/robot/subsystems/boathook/Boathook.java

```diff
@@ -68,68 +75,84 @@ public class Boathook extends SubsystemBase {
                   .withKS(0)
                   .withKG(0)
                   .withGravityType(GravityTypeValue.Arm_Cosine)
-                  .withStaticFeedforwardSign(StaticFeedforwardSignValue.UseVelocitySign));
-  // .withHardwareLimitSwitch(
-  //     new HardwareLimitSwitchConfigs()
-  //         .withForwardLimitEnable(true)
-  //         .withForwardLimitAutosetPositionEnable(true)
-  //         .withForwardLimitAutosetPositionValue(0)
-  //         .withForwardLimitRemoteSensorID(BoathookConstants.BOATHOOK_CANDI_ID)
-  //         .withForwardLimitSource(ForwardLimitSourceValue.RemoteCANdiS1)
-  //         .withForwardLimitType(ForwardLimitTypeValue.NormallyClosed));
+                  .withStaticFeedforwardSign(StaticFeedforwardSignValue.UseVelocitySign))
+          .withHardwareLimitSwitch(
+              new HardwareLimitSwitchConfigs()
+                  .withReverseLimitEnable(true)
+                  .withReverseLimitAutosetPositionEnable(true)
+                  .withReverseLimitAutosetPositionValue(BoathookConstants.ROTATOR_REVERSE_LIMIT)
+                  .withReverseLimitRemoteSensorID(BoathookConstants.CANDI_ID)
+                  .withReverseLimitSource(ReverseLimitSourceValue.RemoteCANdiS1)
+                  .withReverseLimitType(ReverseLimitTypeValue.NormallyOpen)
+                  .withForwardLimitEnable(false)
+                  .withForwardLimitAutosetPositionEnable(false));
+
+  private static final CANcoderConfiguration rotationEncoderConfig =
+      new CANcoderConfiguration()
+          .withMagnetSensor(
+              new MagnetSensorConfigs()
+                  .withMagnetOffset(-0.582) // 0.41
```

Comment:
This looks like the magnet offset of 0 below, with a comment indicating an original value. Might need to check this one too.